sklearn/feature_selection/multivariate_filtering.py

```diff
@@ -0,0 +1,224 @@
+from ..neighbors import NearestNeighbors
+
+
+def _compute_mi_cc(x, y, n_neighbors):
+    n_samples = x.size
+
+    x_std = np.std(x)
+    if x_std > 0:
+        x = x / x_std
+    x = x.reshape((-1, 1))
+
+    y_std = np.std(y)
+    if y_std > 0:
+        y = y / y_std
+    y = y.reshape((-1, 1))
+
+    xy = np.hstack((x, y))
+
+    nn = NearestNeighbors(metric='chebyshev', n_neighbors=n_neighbors)
+
+    nn.fit(xy)
+    radius = nn.kneighbors()[0]
+    radius = np.nextafter(radius[:, -1], 0)
+
+    nn.set_params(algorithm='kd_tree')
+
+    nn.fit(x)
+    ind = nn.radius_neighbors(radius=radius, return_distance=False)
+    nx = np.array([i.size for i in ind])
+
+    nn.fit(y)
+    ind = nn.radius_neighbors(radius=radius, return_distance=False)
+    ny = np.array([i.size for i in ind])
+
+    mi = (digamma(n_samples) + digamma(n_neighbors) -
+          np.mean(digamma(nx + 1)) - np.mean(digamma(ny + 1)))
+
+    return max(0, mi)
+
+
+def _compute_mi_cd(c, d, n_neighbors):
+    c = c.reshape((-1, 1))
+    n_samples = c.size
+
+    nn = NearestNeighbors(n_neighbors=n_neighbors)
+
+    radius = np.empty(n_samples)
+    label_counts = np.empty(n_samples)
+    for label in np.unique(d):
+        mask = d == label
+        count = np.sum(mask)
+        if count > 1:
+            nn.set_params(n_neighbors=min(n_neighbors, count - 1))
+            nn.fit(c[mask])
+            r = nn.kneighbors()[0]
+            radius[mask] = np.nextafter(r[:, -1], 0)
+        else:
+            radius[mask] = 0
+        label_counts[mask] = count
+
+    nn.set_params(algorithm='kd_tree')
+    nn.fit(c)
+    ind = nn.radius_neighbors(radius=radius, return_distance=False)
+    neighbor_counts = np.array([i.size for i in ind])
+
+    mi = (digamma(n_samples) + digamma(n_neighbors) -
+          np.mean(digamma(label_counts)) -
+          np.mean(digamma(neighbor_counts + 1)))
+
+    return max(0, mi)
+
+
+def _compute_mi(x, y, x_categorical, y_categorical, n_neighbors=3):
+    if x_categorical and y_categorical:
+        return mutual_info_score(x, y)
+    elif x_categorical and not y_categorical:
+        return _compute_mi_cd(y, x, n_neighbors)
+    elif not x_categorical and y_categorical:
+        return _compute_mi_cd(x, y, n_neighbors)
+    else:
+        return _compute_mi_cc(x, y, n_neighbors)
+
+
+class MinRedundancyMaxRelevance(BaseEstimator, SelectorMixin):
+    """Select the subset of features based on mutual information criterion.
+
+    The algorithm finds a subset of features jointly having maximum mutual
+    information with the target variable. To avoid estimation of mutual
+    information in high dimensional space the algorithm uses theoretically
+    equivalent incremental search, where the difference between relevance and
+    redundancy of a feature is considered on each iteration [1]_.
+
+    Read more in :ref:`User Guide <MI>`.
+
+    Parameters
+    ----------
+    n_features_to_select : None or int, optional (default=None)
```

Comment:
Perhaps just `limit`? Either way, it should support a float between 0 and 1 to mean a fraction of the total number of features.